src/blockchain/ain/jellyfish/utxo-provider.service.ts

```diff
@@ -155,6 +158,30 @@ export class UtxoProviderService {
     return neededUnspent;
   }
 
+  private static provideUntilAmountPlusFee(
+    unspent: AddressUnspent[],
+    amountPlusFeeBuffer: BigNumber,
+    sizePriority: UtxoSizePriority,
+  ): [AddressUnspent[], BigNumber] {
+    const neededUnspent: AddressUnspent[] = [];
+    let total = new BigNumber(0);
+    let unspentToSearch: AddressUnspent[] = [];
+    unspent = unspent.sort((a, b) =>
+      sizePriority === UtxoSizePriority.BIG ? this.orderDescending(a, b) : this.orderAscending(a, b),
+    );
+    if (sizePriority === UtxoSizePriority.FITTING) {
+      unspentToSearch = unspent.filter((u) => new BigNumber(u.vout.value).gt(amountPlusFeeBuffer));
+    } else {
+      unspentToSearch = unspent;
+    }
```

Comment:
Looks very complicated 😆  perhaps
```ts
    if (sizePriority === UtxoSizePriority.FITTING) {
      unspent = unspent.filter((u) => new BigNumber(u.vout.value).gt(amountPlusFeeBuffer));
    }
```

Reply:
If we filter for FITTING and filter list we have an empty list if requested amount is bigger than biggest UTXO. Therefore I have introduced `unspentToSearch` to keep the ordered list of unspent

Reply:
Interesting, I used the following code
![image](https://user-images.githubusercontent.com/85513542/198531760-26b5799f-8a3e-4846-9624-1996f6dff5ff.png)
and the output was
![image](https://user-images.githubusercontent.com/85513542/198531811-228d30ba-28fb-4ddc-b228-eedd0804ba3a.png)
🤔

Reply:
okay i haven't tested after it got moved into a own function... thought it would be passed per reference... okay then I will change it

Reply:
Yes, that's what I also thought 🤯

Reply:
every day something new to learn... no wonder I can't recall all of the things 😂  I know cheap excuse

Reply:
Me too in this case 🤣